static MAC on VM interfaces for 0.6

phoracek opened the pull request:
**What this PR does / why we need it**:
 Backport of #1213. Depends on #1427 

**Special notes for your reviewer**:
Rebasing of tests was wild, please pay extra attention to that part.

phoracek commented:
Failed functional tests.

cynepco3hahue commented:
ci test please

cynepco3hahue commented:
ci test please

cynepco3hahue commented:
/release-note-none

cynepco3hahue commented:
@phoracek @dankenigsberg @yuvalif I can see all networking tests fail
```
•! Panic in Spec Setup (BeforeEach) [0.049 seconds]
Networking
/root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:48
  VirtualMachineInstance attached to implicit pod network [BeforeEach]
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:407
    should be able to reach
    /root/go/src/kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
      the Inbound VirtualMachineInstance
      /root/go/src/kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

      Test Panicked
      runtime error: index out of range
      /gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/panic.go:505

      Full Stack Trace
      	/gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/panic.go:505 +0x229
      kubevirt.io/kubevirt/tests_test.glob..func16.4.1(0x1143c00, 0x0, 0x0, 0x0)
      	/root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:115 +0x2657
      kubevirt.io/kubevirt/tests_test.glob..func16.4()
      	/root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:186 +0x109
      kubevirt.io/kubevirt/tests.BeforeAll.func1()
      	/root/go/src/kubevirt.io/kubevirt/tests/utils.go:1255 +0x3f
      kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc42058dc20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
      	/root/go/src/kubevirt.io/kubevirt/tests/tests_suite_test.go:42 +0xaa
      testing.tRunner(0xc4204f33b0, 0x1283e10)
      	/gimme/.gimme/versions/go1.10.linux.amd64/src/testing/testing.go:777 +0xd0
      created by testing.(*T).Run
      	/gimme/.gimme/versions/go1.10.linux.amd64/src/testing/testing.go:824 +0x2e0
```

booxter commented on tests/vmi_networking_test.go:
```diff
@@ -99,23 +101,31 @@ var _ = Describe("Networking", func() {
+			inboundVMIWithCustomMacAddress.Labels = map[string]string{"expose": "me"}
+			inboundVMIWithCustomMacAddress.Spec.Subdomain = "myvmi"
+			inboundVMIWithCustomMacAddress.Spec.Hostname = "my-subdomain"
+			inboundVMIWithCustomMacAddress.Spec.Domain.Devices.Interfaces[0].MacAddress = "de:ad:00:00:be:af"
```
You have to call `v1.SetDefaults_NetworkInterface` before accessing the first interface since `NewRandomVMIWithEphemeralDiskAndUserdata` doesn't add it by default.

booxter commented:
`check-patch.el7.x86_64` job fails with:

```
go version go1.10 linux/amd64

invalid value "KUBEVIRT_GIT_VERSION should be a valid Semantic Version\nPlease see more details here: https://semver.org" for flag -ldflags: missing =<value> in <pattern>=<value>

usage: build [-o output] [-i] [build flags] [packages]

Run 'go help build' for details.

make[1]: *** [build] Error 2

make[1]: Leaving directory `/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt'

make: *** [cluster-build] Error 2
```

Is it a known issue?

cynepco3hahue commented:
@booxter IIUC `check-patch.el7.x86_64` does not interesting you, because it includes output from all jobs
Ones that relevant is specific jobs.

dankenigsberg commented:
@cynepco3hahue but what *is* the "interesting" failing test? (I'd love to see if @booxter can help our backporting effort today)

booxter commented:
I'd like to remind that github UI is terrible and doesn't allow for contributors / reviewers to push updates to fix spotted issues. Only the owner can do it. I've pointed out the issue in the functional test suite that I believe contributes to the failures we see in the jobs, but I have no means to move it forward because github sucks.

booxter commented:
@dankenigsberg I've left a comment inline 2 hours ago, not sure what's help in repeating it again. The comment is here: https://github.com/kubevirt/kubevirt/pull/1430#pullrequestreview-145521234

booxter commented:
And for the record, other than the functional test failure, I don't see any obvious issues with the rest of the PR. One could argue in general why we backport new features to stable version branches but that's a separate discussion, and I am not sure I fully grasp our attitude towards release branches in KubeVirt to be of help.

phoracek commented:
@booxter thanks a lot.

I dropped MAC specific tests, it just does not fit to the 0.6 networking tests, or I'm not able to figure out how to get it there. Added static mac to the standard tests. Please do not merge this.

It still fails on one test with:

```
STEP: starting a pod which tries to reach the vmi via the defined service
STEP: waiting for the pod to report a successful connection attempt
level=info timestamp=2018-08-13T10:05:24.344220Z pos=vmi_networking_test.go:69 component=tests msg="[43 43 32 104 101 97 100 32 45 110 32 49 10 43 43 43 32 110 99 32 109 121 115 101 114 118 105 99 101 46 107 117 98 101 118 105 114 116 45 116 101 115 116 45 100 101 102 97 117 108 116 32 49 53 48 48 32 45 105 32 49 32 45 119 32 49 10 78 99 97 116 58 32 67 111 110 110 101 99 116 105 111 110 32 116 105 109 101 100 32 111 117 116 46 10 10 102 97 105 108 101 100 10 43 32 120 61 10 43 32 101 99 104 111 32 39 39 10 43 32 39 91 39 32 39 39 32 61 32 39 72 101 108 108 111 32 87 111 114 108 100 33 39 32 39 93 39 10 43 32 101 99 104 111 32 102 97 105 108 101 100 10 43 32 101 120 105 116 32 49 10]"

------------------------------
• Failure [4.193 seconds]
Networking
/root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:48
  VirtualMachineInstance attached to explicit pod network
  /root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:368
    with a service matching the vmi exposed
    /root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:278
       should be able to reach the vmi based on labels specified on the vmi [It]
      /root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:298

      Expected
          <v1.PodPhase>: Failed
      to equal
          <v1.PodPhase>: Succeeded

      /root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:307
```

phoracek commented:
Oh, can we even test it without underlying bridge?

phoracek commented:
Moved to #1438 and merged.

cynepco3hahue commented:
we need this PR under release-0.6 as well
